src/front/mod.rs

```diff
@@ -14,8 +14,8 @@ use crate::{
 
 /// Helper class to emit expressions
 #[allow(dead_code)]
-#[derive(Default)]
-struct Emitter {
+#[derive(Default, Debug)]
+pub(crate) struct Emitter {
```

Comment:
this doesn't need to be `pub(crate)`. You are using it from `front/glsl`, which is a child of this module